### DISCUSSION TITLE
General purpose dialogue metadata

### DISCUSSION THREAD

**Discussion by Selenyhr** [nathanhoad/godot_dialogue_manager#287] [Ideas]
Hey Nathan!

So thinking back about #272, while what I suggested should work, it may not be the most elegant solution, since it requires extra custom processing on the character side.
Do you think it would be possible to add a general-purpose user-defined metadata piece of information to dialogues and responses, akin to the current `[ID:]` tag, but where the user could write any kind of data?

Maybe something like so:
```
Nathan: Hello there! [META:wave] [ID:uhgr85zef]
```

And then within the dialogue line, you could get the raw string that is passed right after the `[META:]` tag, like you did the the id in ebaadefa3caad08119a996c68126d79d93ad48c5. Of course, this piece of data wouldn't be translated in any way.

This shouldn't be too hard to implement, and would allow general metadata to be used for stateful games, all the while Dialogue Manager remaining stateless.

Thanks for your feedback :)

**Comment by nathanhoad** [comment]
I have [a PR in for adding tags](https://github.com/nathanhoad/godot_dialogue_manager/pull/288) which should be useful for most things. For anything else there's always mutations like `[do set_thing("something")]`.

**Comment by Selenyhr** [comment]
Perfect! That's exactly the same kind of idea, just with a different syntax :) thank you very much!
